test/silhouette.jl

```diff
@@ -44,11 +44,14 @@ c = [3, 1]
     d[1, 2] = d[2, 1] = 5
     
     @test silhouettes(a, d) ≈ [[-0.5, -0.5]; fill(0.0, 8)]
+end
 
+@testset "silhouette() throws an error when degenerated clustering is given" begin
     a = fill(1, 10)
-
+    d = fill(1, (10, 10))
+    [d[i, i] for i=1:10]
```

Comment:
At the moment this line does nothing, but it's also not needed since the test only takes `a` into account.
In general I'd recommend to use normal `for i in 1:10; d[i, i] = 0; end` loop instead of a comprehension with a side effect.